SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+This is especially critical for large arrays, supplied externally, being passed through where we do not want to do a copy operation before entering C++ and also do not want to normalize units for millions of data points at that time.
+Large data arrays and, in particular, table columns returned from C++ layers would benefit from being returned as quantities, as this gives much more flexibility to users of the data and minimizes the risk of misinterpretation of values if units are later changed for a column or the user relies entirely on documentation to interpret values.
+The proposal is that LSST software adopts quantities on a case-by-case basis and are continuously on the look out for  any performance impact associated with quantities.
+
+\section{Proposal}
+
+The concrete proposals from this investigation are therefore:
+
+\begin{itemize}
+\item Change EUPS packaging to use a \texttt{setup.py} and upload it to PyPI.
+\item Investigate the addition of a \texttt{setup.py} for standard LSST Python packages in addition to our current Scons-based build system.
+\item Support export of LSST tables to Astropy tables.
+\item Determine whether STC-2 is a suitable basis for WCS interoperability.
+\item Remove LSST astronomical coordinate classes and replace with Astropy coordinates in Python and a generic spherical coordinates class in C++.
+\item Investigate migration away from SWIG for C++ bindings.
+\item Contribute the LSST \emph{footprint} code to Astropy.
+\item Create an Astropy affiliated package from the AFW detection package.
+\item Investigate rationalizing of LSST and Astropy spherical geometry region handling.
+\item Modify \texttt{Exposure} to implement an \texttt{NDData}-like interface.
+\item Work with the Astropy project to add required functionality such as pixel origins to \texttt{NDData}.
+\item LSST will allow Astropy quantity objects to be used in public interfaces on a case by case basis.
+
+\end{itemize}
+
+With these changes implemented LSST will have a solid base from which to assess whether it is feasible to migrate more LSST code to the community as affiliated packages.
+It is probable that LSST will start to migrate the existing codebase entirely to Astropy classes where appropriate classes exist, and in particular adopt \texttt{astropy.time} and \texttt{astropy.coordinates} as soon as possible.
+
+\section{Conclusion}
+
+The LSST project has a code base that has developed over a number of years and has been developed independently of the wider Python astronomy software community.
+We have looked at the Astropy package and worked with the Astropy team to understand areas where synergies exist and where LSST can contribute more fully to the community without delaying our primary deliverables.
+The proposals outlined in this document go a long way towards integrating LSST into the wider software community and should bring key benefits to LSST and research astronomers.
```

Comment:
Is it worth adding a few words about exactly what the status of this work is? Given that the preponderance of the authors are working for LSST, I think the reader could legitimately be confused as to whether we're talking about blue-skies proposals or work we're undertaking to carry out or something in between.